pkg/network/tcp_peer.go

```diff
@@ -113,7 +114,7 @@ func (p *TCPPeer) BroadcastHPPacket(ctx context.Context, msg []byte) error {
 // putMsgIntoQueue serializes the given Message and puts it into given queue if
 // the peer has done handshaking.
 func (p *TCPPeer) putMsgIntoQueue(queue chan<- []byte, msg *Message) error {
-	b, err := msg.Bytes()
+	b, err := msg.BytesCompressed(p.SupportsCompression())
```

Comment:
I'm a little bit worried about this change because subsequent `putPacketIntoQueue` makes call to `p.handshaked()` whereas `p.SupportsCompression()` performs the same call. So may be we need additional helper, what do you think?

Reply:
`SupportsCompression` can just not look into `handshaked()` at all. Start with compression disabled, enable it after `version` exchange and that's it.

Reply:
Fixed.